File: engine/engine-model/src/main/scala/tools/packet/cashshop/ShowCash.scala
package tools.packet.cashshop

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowCash(private var _nxCredit: Int, private var _maplePoint: Int, private var _nxPrepaid: Int) extends PacketInput {
  def nxCredit: Int = _nxCredit

  def maplePoint: Int = _maplePoint

  def nxPrepaid: Int = _nxPrepaid

  override def opcode(): SendOpcode = SendOpcode.QUERY_CASH_RESULT
}

File: engine/engine-model/src/main/java/tools/packet/cashshop/ShowCash.java
package tools.packet.cashshop;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowCash(Integer nxCredit, Integer maplePoint, Integer nxPrepaid) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.QUERY_CASH_RESULT;
   }
}